internal/funcs/0.12/base_functions.go

```diff
@@ -0,0 +1,994 @@
+package funcs
+
+import (
+	"github.com/hashicorp/hcl-lang/schema"
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/function"
+)
+
+func BaseFunctions() map[string]schema.FunctionSignature {
+	return map[string]schema.FunctionSignature{
+		"abs": {
+			Params: []function.Parameter{
+				{
+					Name: "num",
+					Type: cty.Number,
+				},
+			},
+			ReturnType:  cty.Number,
+			Description: "`abs` returns the absolute value of the given number. In other words, if the number is zero or positive then it is returned as-is, but if it is negative then it is multiplied by -1 to make it positive before returning it.",
+		},
+		"base64decode": {
+			Params: []function.Parameter{
+				{
+					Name: "str",
+					Type: cty.String,
+				},
+			},
+			ReturnType:  cty.String,
+			Description: "`base64decode` takes a string containing a Base64 character sequence and returns the original string.",
+		},
+		"base64encode": {
+			Params: []function.Parameter{
+				{
+					Name: "str",
+					Type: cty.String,
+				},
+			},
+			ReturnType:  cty.String,
+			Description: "`base64encode` applies Base64 encoding to a string.",
+		},
+		"base64gzip": {
+			Params: []function.Parameter{
+				{
+					Name: "str",
+					Type: cty.String,
+				},
+			},
+			ReturnType:  cty.String,
+			Description: "`base64gzip` compresses a string with gzip and then encodes the result in Base64 encoding.",
+		},
+		"base64sha256": {
+			Params: []function.Parameter{
+				{
+					Name: "str",
+					Type: cty.String,
+				},
+			},
+			ReturnType:  cty.String,
+			Description: "`base64sha256` computes the SHA256 hash of a given string and encodes it with Base64. This is not equivalent to `base64encode(sha256(&amp;#34;test&amp;#34;))` since `sha256()` returns hexadecimal representation.",
```

Comment:
I'm not sure where this came from and whether it affects other places in Core as well, but I see some HTML escaping in effect (`&amp;#...`) which probably shouldn't be.

The context in which we need to escape the (Markdown) description text would be single-line Go string, i.e. `"` would need to be `\"` but that's about it. Single quotes and probably most other characters can stay as-is.

I don't know if there's any escaping function we could use in this context in the generator, here I suppose it's just a simple matter of manual Find+Replace.

Reply:
Good catch! I fixed it in Terraform in https://github.com/hashicorp/terraform/pull/32710 and replaced all occurrences here.